### PR TITLE
add 'supported'-label to all supported apps

### PR DESCRIPTION
**Pull request by schiessle**
At the moment we add the "supported"-label only to apps which are already downloaded to the apps folder.

As a user with a valid Nextcloud subscription I would like to see directly which app is supported by my subscription, also if it is not yet downloaded from app.nextcloud.com.

This PR also check apps from apps.nextcloud.com against the list of supported apps and add the "supported" label if the app is covered by the users subscription.

### CODE REVIEW

**Comment by schiessle** [comment]
php-cs check warnings seems to be unrelated to my changes

File: apps/settings/lib/Controller/AppSettingsController.php [hunk 1]
```diff
@@ -204,13 +204,19 @@ private function fetchApps() {
...
 			if (!array_key_exists($app['id'], $this->allApps)) {
 				$this->allApps[$app['id']] = $app;
 			} else {
 				$this->allApps[$app['id']] = array_merge($app, $this->allApps[$app['id']]);
 			}
+
+			if (in_array($app['id'], $supportedApps)) {
+				$this->allApps[$app['id']]['level'] = \OC_App::supportedApp;
+			}
+
```
Comment (MichaIng) [inline apps/settings/lib/Controller/AppSettingsController.php:219]: This space here might be responsible for the `php-cs` violations, which **are** related to these changes 😉. It is a bit misleading, but a `braces` violation always implies a `class_definition` violation for some reason, I recognised this in a few other cases: Solving the `braces` issue solves the `class_definition` as well.

Comment (schiessle) [reply apps/settings/lib/Controller/AppSettingsController.php:219]: Thanks for the hint... Let's try it, I removed the empty line.

Comment (MichaIng) [reply apps/settings/lib/Controller/AppSettingsController.php]: That worked, `php-cs` is green now 👍.

Comment (MichaIng) [reply apps/settings/lib/Controller/AppSettingsController.php]: One of these acceptance tests fails in ~50% of cases without a specific reason. Rebasing the PR likely solves it, or a different acceptance test fails, which is then a prove that it is unrelated. I would love if those were more reliable, as it is a pain to have merged blocked without running through the whole CI again.

Comment (MichaIng) [reply apps/settings/lib/Controller/AppSettingsController.php]: The one failing acceptance files app is solved here: #29081

**Comment by schiessle** [comment]
All tests passed 🚀 Would be really nice to get this into Nc 23 :slightly_smiling_face:

**Review by MichaIng — APPROVED** [review]

**Review by CarlSchwan — APPROVED** [review]
Looks good to me :)

**Pull request merged**